src/binder/bind/bind_ddl.cpp

```diff
@@ -50,14 +54,42 @@ unique_ptr<BoundStatement> Binder::bindDropTable(const Statement& statement) {
     auto tableName = dropTable.getTableName();
     validateTableExist(catalog, tableName);
     auto catalogContent = catalog.getReadOnlyVersion();
-    auto isNodeTable = catalogContent->containNodeTable(tableName);
     auto tableID = catalogContent->getTableID(tableName);
-    if (isNodeTable) {
+    if (catalogContent->containNodeTable(tableName)) {
         validateNodeTableHasNoEdge(catalog, tableID);
     }
     return make_unique<BoundDropTable>(tableID, tableName);
 }
 
+unique_ptr<BoundStatement> Binder::bindRenameTable(const Statement& statement) {
+    auto renameTable = (RenameTable&)statement;
+    auto tableName = renameTable.getTableName();
+    auto catalogContent = catalog.getReadOnlyVersion();
+    validateTableExist(catalog, tableName);
+    if (catalogContent->containNodeTable(renameTable.getNewName()) ||
+        catalogContent->containRelTable(renameTable.getNewName())) {
+        throw BinderException("Table: " + renameTable.getNewName() + " already exists.");
+    }
+    return make_unique<BoundRenameTable>(
+        catalogContent->getTableID(tableName), tableName, renameTable.getNewName());
+}
+
+unique_ptr<BoundStatement> Binder::bindAddProperty(const Statement& statement) {
+    auto& addProperty = (AddProperty&)statement;
+    auto tableName = addProperty.getTableName();
+    validateTableExist(catalog, tableName);
+    auto catalogContent = catalog.getReadOnlyVersion();
+    auto tableID = catalogContent->getTableID(tableName);
+    auto dataType = Types::dataTypeFromString(addProperty.getDataType());
+    if (catalogContent->getTableSchema(tableID)->containProperty(addProperty.getPropertyName())) {
+        throw BinderException("Property: " + addProperty.getPropertyName() + " already exists.");
+    }
+    return make_unique<BoundAddProperty>(tableID, addProperty.getPropertyName(), dataType,
+        ExpressionBinder::implicitCastIfNecessary(
```

Comment:
I suggest extract ExpressionBinder::implicitCastIfNecessary into its own line with a local variable.

Reply:
extracted the statement and made it a local variable